runtime/codert_vm/cnathelp.cpp

```diff
@@ -3853,7 +3858,9 @@ old_slow_jitReportStaticFieldRead(J9VMThread *currentThread)
 	J9JavaVM *vm = currentThread->javaVM;
 	void *addr = NULL;
 
-	if (J9_EVENT_IS_HOOKED(vm->hookInterface, J9HOOK_VM_GET_STATIC_FIELD)) {
+	if (J9_EVENT_IS_HOOKED(vm->hookInterface, J9HOOK_VM_GET_STATIC_FIELD)
+		&& SHOULD_DISPATCH_HOOK(currentThread, NULL)
```

Comment:
In the future, if one of selected events is triggered from a new location without `SHOULD_DISPATCH_HOOK`, then it will be easy to introduce a bug.

Is there a common/centralized location where we can perform the `SHOULD_DISPATCH_HOOK` check for all the selected J9 hook events?
```
hookable.cpp::J9HookDispatch(struct J9HookInterface **hookInterface, uintptr_t taggedEventNum, void *eventData) {
   if (taggedEventNum == one of the events listed in j9.h's comment)
   && SHOULD_DISPATCH_HOOK(currentThread, NULL)
   ) {
```

Reply:
Another way is to modify the `J9_EVENT_IS_HOOKED` macro but I'm not sure if we have access to current thread in those locations.

If we want a openj9-specific solution we might just wrap the whole condition inside a helper function.

Which way do you prefer?

Reply:
Please do not modify the general hook library for this.

Reply:
Resolving, let's keep it as-is.